MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -9,240 +15,336 @@ namespace MediaBrowser.Common.Net
+        /// <param name="negated">Boolean signaling if negated or not negated values should be parsed.</param>
+        /// <returns><c>True</c> if parsing was successful.</returns>
+        public static bool TryParseToSubnets(string[] values, [NotNullWhen(true)] out IReadOnlyList<IPNetwork>? result, bool negated = false)
         {
-            if (dest is null || source.Count != dest.Count)
+            if (values is null || values.Length == 0)
             {
+                result = null;
                 return false;
             }
 
-            foreach (var sourceItem in source)
+            var tmpResult = new List<IPNetwork>();
+            for (int a = 0; a < values.Length; a++)
             {
-                bool found = false;
-                foreach (var destItem in dest)
-                {
-                    if (sourceItem.Equals(destItem))
-                    {
-                        found = true;
-                        break;
-                    }
-                }
-
-                if (!found)
+                if (TryParseToSubnet(values[a], out var innerResult, negated))
                 {
-                    return false;
+                    tmpResult.Add(innerResult);
                 }
             }
 
-            return true;
+            result = tmpResult;
+            return tmpResult.Count > 0;
         }
 
         /// <summary>
-        /// Returns a collection containing the subnets of this collection given.
+        /// Try parsing a string into an <see cref="IPNetwork"/>, respecting exclusions.
+        /// Inputs without a subnet mask will be represented as <see cref="IPNetwork"/> with a single IP.
         /// </summary>
-        /// <param name="source">The <see cref="Collection{IPObject}"/>.</param>
-        /// <returns>Collection{IPObject} object containing the subnets.</returns>
-        public static Collection<IPObject> AsNetworks(this Collection<IPObject> source)
+        /// <param name="value">Input string to be parsed.</param>
+        /// <param name="result">An <see cref="IPNetwork"/>.</param>
+        /// <param name="negated">Boolean signaling if negated or not negated values should be parsed.</param>
+        /// <returns><c>True</c> if parsing was successful.</returns>
+        public static bool TryParseToSubnet(ReadOnlySpan<char> value, [NotNullWhen(true)] out IPNetwork? result, bool negated = false)
         {
-            ArgumentNullException.ThrowIfNull(source);
-
-            Collection<IPObject> res = new Collection<IPObject>();
-
-            foreach (IPObject i in source)
+            var splitString = value.Trim().Split('/');
+            if (splitString.MoveNext())
             {
-                if (i is IPNetAddress nw)
+                var ipBlock = splitString.Current;
+                var address = IPAddress.None;
+                if (negated && ipBlock.StartsWith("!") && IPAddress.TryParse(ipBlock[1..], out var tmpAddress))
```

Comment:
```suggestion
                if (negated && ipBlock.StartsWith('!') && IPAddress.TryParse(ipBlock[1..], out var tmpAddress))
```

Reply:
This does not work: `cannot convert from 'char' to 'System.ReadOnlySpan<char>'`